quinn-proto/src/crypto.rs

```diff
@@ -64,8 +64,14 @@ impl CryptoSession for TlsSession {
         }
     }
 
-    fn read_handshake(&mut self, buf: &[u8]) -> Result<(), TLSError> {
-        self.read_hs(buf)
+    fn read_handshake(&mut self, buf: &[u8]) -> Result<(), TransportError> {
+        self.read_hs(buf).map_err(|_| {
+            if let Some(code) = self.error_code() {
```

Comment:
I think we can remove `error_code` from the trait entirely with this change.

Reply:
Right! Fixed.